File: src/main/resources/static/src/app/models/Practitioner.ts
export class Practitioner {
  constructor(
    public id: string = "",
    public resourceType: string = "Practitioner",
    public identifier?: Array<Identifier>,
    public name?: Array<HumanName>,
    public telecom?: Array<ContactPoint>,
    public active: boolean = false,
    public gender: string = "unknown",
    public birthDate: Date = new Date(1000, 1, 1),
    public address?: Array<Address>,
    public photos?: Array<Attachment>,
    public qualification?: Array<Qualification>,
  ) {}
}
export class HumanName {
  constructor(
    public id: string = "",
    public use: string = "",
    public text: string = "",
    public family: string = "",
    public given: string = "",
    public suffix: string = ""
  ) {}
}
export class Identifier {
  constructor(
    public value: string = ""
  ) {}
}
export class ContactPoint {
  constructor() {}
}
export class Address {
  constructor() {}
}
export class Attachment {
  constructor(
    public data: string = "",
    public contentType: string = ""
  ) {}
}
export class Qualification{
    constructor(
      public id: string = "",
      public code: CodableConcept,
      public period: Period,
      public issuer: string = "",
    ) {}
}
export class Period{
  constructor(
    public id: string = "",
    public start: string = ""
  ) {}
}
export class CodableConcept{
  constructor (
    public id: string = "",
    public text: string = ""
  ){}
}

File: src/main/java/at/spengergasse/fhirstarter/entity/Practitioner.java
package at.spengergasse.fhirstarter.entity;

import at.spengergasse.fhirstarter.model.*;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name="pr_practitioner")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class Practitioner extends DomainResource {

    public boolean isActive() {
        return active;
    }

    public enum GenderCode{
        male, female, other, unknown
    }
    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "i_pr_id", referencedColumnName = "id")
    private List<Identifier> identifier;

    @Column(name = "pr_active")
    private boolean active;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "hn_pr_id", referencedColumnName = "id")
    private List<HumanName> name;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "cp_pr_id", referencedColumnName = "id")
    private List<ContactPoint> telecom;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "a_pr_id", referencedColumnName = "id")
    private List<Address> address;

    @Enumerated(EnumType.STRING)
    @Column(name = "pr_gender")
    private GenderCode gender;

    @Column(name = "pr_birthdate")
    private LocalDate birthDate;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "at_pr_id", referencedColumnName = "id")
    private List<Attachment> photo;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "q_pr_id", referencedColumnName = "id")
    private List<Qualification> qualification;
}
